openerp/addons/base/ir/ir_qweb.py

```diff
@@ -432,6 +432,30 @@ def render_tag_if(self, element, template_attributes, generated_attributes, qweb
 
     def render_tag_call(self, element, template_attributes, generated_attributes, qwebcontext):
         d = qwebcontext.copy()
+
+        # we need 1/ a valid context 2/ a t-lang tag with the t-call
+        translation_poss = d.context and 'lang' in d.context and 'lang' in template_attributes
+
+        # if there is a t-lang attribute, change context to include this language
+        if translation_poss:
+            # we need to store the original language
+            d.context['init_lang'] = d.context['lang']
+            lang = template_attributes['lang']
+            # lang can be a lang code (ex: 'en_US' or 'fr') or a record field
+            lang_regex = re.compile("^[a-z]{2}(_[A-Z]{2})?$")
+            if not lang_regex.match(lang):
+                try:
+                    try_lang=self.eval(lang,d)
+                    if not try_lang or not lang_regex.match(try_lang):
+                        raise_qweb_exception()
+                    lang = try_lang
+                except QWebException:
+                    _logger.warning("Could not render element %s", lang)
+                    _logger.warning(" Using default language %s", d.context['lang'])
+                    lang = d.context['lang']
+                    pass
+            d.context['lang']= lang
+
```

Comment:
if I understand correctly: if lang is already a language code, it is ok, else you try to eval it, and if it did not work, you raise a warning... am I right?